.github/workflows/coverage.yml

```diff
@@ -67,7 +67,9 @@ jobs:
           source deactivate_conanrun.sh
           lcx="lcov --output-file=coverage.info " && for i in `find . -name "*.info.cleaned"`; do lcx+=" --add-tracefile=$i"; done && $lcx
       - name: Codecov
-        uses: codecov/codecov-action@ab904c41d6ece82784817410c45d8b8c02684457
+        uses: codecov/codecov-action@v4
```

Comment:
per advise of the Apache Infra team, please use the hash/checksum here instead of a tag.

also see https://github.com/apache/celix/commit/2ca790de93a9fb42a14c783aa726ff40921683fa were we did this for the previous version